app.py

```diff
@@ -75,104 +79,156 @@ def voinfo():
     vos_xml = to_xml(vos)
     return Response(vos_xml, mimetype='text/xml')
 
-@app.route('/rgsummary/xml')
-def resources():
-    global _rgsummary, _rgdowntime
-    if not _rgsummary:
-        _rgsummary, _rgdowntime = get_rgsummary_rgdowntime()
 
-    rgsummary = copy.deepcopy(_rgsummary)
-    rgs = rgsummary["ResourceSummary"]["ResourceGroup"]
-    args = flask.request.args
+
+
+def get_filters_from_args(args) -> Filters:
```

Comment:
Whoa wait, are we using Python3 for this project?

Reply:
Yes.